guides/managing-current-user.md

```diff
@@ -72,6 +72,50 @@ export default Ember.Service.extend({
 });
 ```
 
+#### Using a dedicated endpoint and returning actual user id in response
+
+```js
+// app/services/current-user.js
+import Ember from 'ember';
+
+const { inject: { service }, isEmpty, RSVP } = Ember;
+
+export default Ember.Service.extend({
+  session: service('session'),
+  store: service(),
+
+  load() {
+    if (this.get('session.isAuthenticated')) {
+      return this.get('store').queryRecord('user', { me: true }).then((user) => {
+        this.set('user', user);
+      });
+    }
+  }
+});
+
+// app/adapters/user.js
+import ApplicationAdapter from './application';
+
+export default ApplicationAdapter.extend({
+  urlForQueryRecord(query) {
+    if (query.me) {
+      delete query.me;
+      return `${this._super(...arguments)}/me`;
+    }
+
+    return this._super(...arguments);
+  }
+});
+```
+
+If the `find` method is called with id `me` and the response includes the actual
+user id, ember data will create an empty record with id `me` and all other model
+attributes empty.
+
+Using the `queryRecord` method and overriding the adapter avoid this problem.
+The adapter override will generate `api/users/me` when the `me` query param is
+present.
```

Comment:
Let's change the ending to something like this:

... when `store.queryRecord()` is invoked with a query where the `me` param is present.